src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -791,10 +791,7 @@ private void ConfigureDefaultLoggerFactory()
             scriptConfig.HostConfig.LoggerFactory.AddProvider(new FileLoggerProvider(traceWriteFactory,
                 (category, level) => (category == LogCategories.Function) && isFileLoggingEnabled()));
 
-            // Allow a way to plug in custom LoggerProviders.
-            ILoggerFactoryBuilder builder = scriptConfig.HostConfig.GetService<ILoggerFactoryBuilder>() ??
-                new DefaultLoggerFactoryBuilder();
-            builder.AddLoggerProviders(scriptConfig.HostConfig.LoggerFactory, scriptConfig, settingsManager);
+            scriptConfig.LoggerFactoryBuilder.AddLoggerProviders(scriptConfig.HostConfig.LoggerFactory, scriptConfig, settingsManager);
```

Comment:
So LoggerFactory extensibility is in core SDK, but factory builder extensibility is only in Script? Why is that out of curiosity? What if someone (e.g. CLI) wants to plug in both - it does so in two different locations?

Reply:
Script needs this `Builder` b/c we have to construct a `LoggerFactory` over-and-over again whenever the host restarts (since it's built based on host.json). In SDK, there's no real need for that. You can just build the LoggerFactory on your own and pass it to us -- you only ever need to do it once.

Reply:
Sorry, didn't answer your other question -- CLI will pass us in a custom `Builder` with console logging and we'll use that to construct the `LoggerFactory`.

Reply:
I do find all these builders/factories hard to understand, but likely I don't have all the context.